azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchImpl.java

```diff
@@ -0,0 +1,414 @@
+import java.lang.Thread.State;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is used to dispatch execution on Containerized infrastructure. Each execution will
+ * be dispatched in single container. This way, it will bring isolation between all the
+ * executions.
+ */
+@Singleton
+public class ContainerizedDispatchImpl extends AbstractExecutorManagerAdapter {
+
+  private final ContainerizedImpl containerizedImpl;
+  private QueueProcessorThread queueProcessor;
+  private final RateLimiter rateLimiter;
+  private static final Logger logger = LoggerFactory.getLogger(ContainerizedDispatchImpl.class);
+
+  @Inject
+  public ContainerizedDispatchImpl(final Props azkProps, final ExecutorLoader executorLoader,
+      final CommonMetrics commonMetrics, final ExecutorApiGateway apiGateway,
+      final ContainerizedImpl containerizedImpl) throws ExecutorManagerException{
+    super(azkProps, executorLoader, commonMetrics, apiGateway);
+    rateLimiter =
+        RateLimiter.create(azkProps.getInt(ContainerizedExecutionManagerProperties.CONTAINERIZED_CREATION_RATE_LIMIT, 20));
+    this.containerizedImpl = containerizedImpl;
+  }
+
+  /**
+   * Get execution ids of all running (unfinished) flows from database.
+   */
+  public List<Integer> getRunningFlowIds() {
+    final List<Integer> allIds = new ArrayList<>();
+    try {
+      getExecutionIdsHelper(allIds, this.executorLoader.fetchUnfinishedFlows().values());
+    } catch (final ExecutorManagerException e) {
+      this.logger.error("Failed to get running flow ids.", e);
+    }
+    return allIds;
+  }
+
+  /**
+   * Get queued flow ids from database. The status for queued flows is READY for
+   * containerization.
+   * @return
+   */
+  public List<Integer> getQueuedFlowIds() {
+    final List<Integer> allIds = new ArrayList<>();
+    try {
+      getExecutionIdsHelper(allIds, this.executorLoader.fetchQueuedFlows(Status.READY));
+    } catch (final ExecutorManagerException e) {
+      this.logger.error("Failed to get queued flow ids.", e);
+    }
+    return allIds;
+  }
+
+  /**
+   * Get size of queued flows. The status for queued flows is READY for containerization.
+   * @return
+   */
+  @Override
+  public long getQueuedFlowSize() {
+    return getQueuedFlowIds().size();
+  }
+
+  /**
+   * Get dispatch method enum for this class.
+   * @return
+   */
+  @Override
+  public DispatchMethod getDispatchMethod() {
+    return DispatchMethod.CONTAINERIZED;
+  }
+
+  /**
+   * This method is used to start queue processor thread. The queue processor thread will pick
+   * execution from queue maintained in database and create a container for it.
+   */
+  @Override
+  public void start() {
```

Comment:
start is too generic, perhaps, startQueueProcessor?

Reply:
It is basically to start dispatch. Later on this method can have other thread processor as well. That is why it is kept generic. Please let me know if you think otherwise.